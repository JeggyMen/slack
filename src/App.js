import React, { useState } from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import Login from './Component/Login/Login';
import SideBar from "./Component/SideBar/SideBar";
import ChatWindow from "./Component/ChatWindow/ChatWindow";
import Signup from "./Component/Signup/Signup";
import './App.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import MainPage from './MainPage/MainPage';
const App = () => {
  const [isLoggedIn, setIsLoggedIn] = useState(false);
  const [user, setUser] = useState(null); 
 
  const handleLogin = (user) => {
    setUser(user);
    setIsLoggedIn(true);
  };
  
  const handleLogout = () => {
    setUser(null);
    setIsLoggedIn(false);
  };
  return (
    <div className="App">
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Login handleLogin={handleLogin} />} />
          <Route path="/Login" element={<Login handleLogin={handleLogin} />} />
          <Route path="/Signup" element={<Signup />} />
          <Route path="/MainPage" element={<MainPage setIsLoggedIn={setIsLoggedIn} user={user} />} />
        </Routes>
      </BrowserRouter>
      {isLoggedIn && user (
        <div>
          <nav>
            <SideBar handleLogout={handleLogout} />
          </nav>
          <main>
            <ChatWindow user={user} />
          </main>
        </div>
      )}
    </div>
  );
};
export default App;