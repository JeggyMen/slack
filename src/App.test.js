import { render, screen } from '@testing-library/react';
import renderer from "react-test-renderer";
import App from './App';

test('renders App.js', () => {
  render(<App />);
});

test("snapshot test for App.js", () => {
  const counter = renderer.create(<App />).toJSON();
  expect(counter).toMatchSnapshot();
});
