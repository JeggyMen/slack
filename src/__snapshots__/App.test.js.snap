// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for App.js 1`] = `
<div
  className="App"
>
  <div
    className="row g-0 vh-100 justify-content-center align-items-center login-container"
  >
    <div
      className="col-10 row g-0 align-items-center border rounded-2 border-shadow bg-white"
    >
      <div
        className="d-none d-md-block col-6"
      >
        <img
          alt="Description of image"
          className="image-small"
          src="/assets/images/slack1.png"
        />
      </div>
      <form
        className="col-12 col-md-6 py-4 px-3"
        onSubmit={[Function]}
      >
        <h4
          className="login-title text-center py-2 mb-2"
        >
          Login
        </h4>
        <div
          className="form-floating mb-3"
        >
          <input
            className="form-control"
            id="email"
            onChange={[Function]}
            placeholder="name@example.com"
            required={true}
            type="email"
          />
          <label
            htmlFor="email"
          >
            Email
          </label>
        </div>
        <div
          className="form-floating mb-3"
        >
          <input
            className="form-control"
            id="password"
            onChange={[Function]}
            placeholder="password"
            required={true}
            type="password"
          />
          <label
            htmlFor="password"
          >
            Password
          </label>
        </div>
        <div
          className="text-center"
        >
          <button
            className="login-btn py-3 rounded-3"
            type="submit"
          >
            Login
          </button>
        </div>
        <div
          className="text-center mt-4"
        >
          Not Registered? 
          <a
            href="/Signup"
            onClick={[Function]}
          >
            Sign Up
          </a>
        </div>
      </form>
    </div>
  </div>
</div>
`;
